oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedMongoDownloadTask.java

```diff
@@ -547,15 +547,9 @@ private Future<?> submitDownloadTask(ExecutorCompletionService<Void> executor, D
             try {
                 downloadTask.download(mongoFilter);
                 downloadTask.reportFinalResults();
-            } catch (InterruptedException | MongoInterruptedException e) {
-                LOG.info("Thread interrupted: {}", e.toString());
+            } catch (Throwable e) {
+                LOG.warn("Error during download: {}", e.toString());
```

Comment:
Technically, with InterruptedException we should set the interrupt flag?

Reply:
My understanding is that we only need to set the interrupt flag again after catching an InterruptedException if we do not propagate an exception that signals that the thread should stop. In this particular case, the code throws a RuntimeException, wrapping the InterruptedException, which will stop the task. So the interrupt already fulfilled its task, which was to stop the thread.